File: FSharp/ProjectEuler5.fs
let listNumber = [2..20]

let rec findSmallestNumberDividedByWholeNumberFrom1To20 n listNumber=
    if  listNumber |> List.exists(fun x -> n % x <> 0) then
         findSmallestNumberDividedByWholeNumberFrom1To20 (n + 1) listNumber
    else
         n

[<EntryPoint>]
let main argv = 
    let result = findSmallestNumberDividedByWholeNumberFrom1To20 2 listNumber
    printfn "%A" result
    0 // return an integer exit code

File: TypeScript/ProjectEuler5.ts
class List {
    private list: number[] = [];

    constructor(to:number) {
        for (var i = 1; i < to + 1; i++) {
                this.list.push(i);
        }
    }

    getList() {
        return this.list;
    }
}

var findSmallestNumberDividedByWholeNumberFrom1To20  = (n: number, listNumber: number[]) => {
    if (listNumber.filter(x=>n % x !== 0).length > 0) {
        return findSmallestNumberDividedByWholeNumberFrom1To20(n + 1, listNumber);
    } else {
        return n;
    }
}
var list = new List(20).getList();
console.log(findSmallestNumberDividedByWholeNumberFrom1To20(2,list));
while (true) {
    
}
